admin/server/organizations.go

```diff
@@ -471,3 +536,40 @@ func organizationToDTO(o *database.Organization) *adminv1.Organization {
 		UpdatedOn:   timestamppb.New(o.UpdatedOn),
 	}
 }
+
+func paginationOptions(reqToken string, pageSize int) (*database.PaginationOptions, error) {
+	opts := &database.PaginationOptions{PageSize: validPageSize(pageSize)}
+	if reqToken != "" {
+		token := &adminv1.PageToken{}
+		in, err := base64.URLEncoding.DecodeString(reqToken)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := proto.Unmarshal(in, token); err != nil {
+			return nil, fmt.Errorf("Failed to parse request token: %w", err)
+		}
+		opts.Cursor = token.Cursor[0]
+	}
+	return opts, nil
+}
+
+func nextPageToken(cursor string) (string, error) {
+	token := &adminv1.PageToken{Cursor: []string{cursor}}
+	bytes, err := proto.Marshal(token)
+	if err != nil {
+		return "", err
+	}
+
+	return base64.URLEncoding.EncodeToString(bytes), nil
+}
+
+func validPageSize(pageSize int) int {
+	if pageSize <= 0 {
+		return _defaultPageSize
+	}
+	if pageSize > _maxPageSize {
+		return _maxPageSize
+	}
+	return pageSize
+}
```

Comment:
1. Replace `paginationOptions` with `unmarshalPageToken(token string) (*adminv1.PageToken, error)`
2. Remove `validPageSize` and validate using `(validate.rules)` annotations in the proto file instead.
3. Replace `nextPageToken` with something like `marshalPageToken(val string) string`. There's no need to return an `error` here, it can just `panic` (would be a dev error).
4. Move this code to `server.go` or a `page_token.go` file, this is not specific to organizations

Reply:
2. Wouldn't that mean that we will reject the request if `page_size` > max. Is there a way in `validate.rules` to change the value if `gt` some value ?

Reply:
2. It's best to just reject the request instead of silently changing behavior. It would be discovered at frontend dev time anyway, so they could fix it immediately.

Reply:
From what I understood this is definitely not a best practice to reject the request in this case and a server may always return less records than what has been requested

Reply:
Page token invalid and page size higher than limit are totally different cases. The handling needs to be looked on a case to case basis.
Github APIs return response capping to max limit even if higher page size is requested for example : 
```
curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer <token>"\
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/OWNER/REPO/issues?per_page=100000
```
Not returning an error and limiting to max page size would allow us to reduce these max limits in future if required without breaking the client.